homepage/tests/courses_methods_tests.py

```diff
@@ -0,0 +1,198 @@
+import pytest
+from homepage.models import Course
+from django.core.exceptions import ValidationError
+from decimal import Decimal
+
+# -----------course tests----------- #
+
+
+# creating a single new course with valid input
+@pytest.mark.parametrize("valid_courses", [
+    (1, "Linear Algebra 1", True, 4),
+    #                               ^- no input at all for syllabi onwards
+    (1, "Linear Algebra 1", True, 4, None),
+    #                                    ^- no input at all for avg_rating onwards
+    (1, "Linear Algebra 1", True, 4, "https://www.google.com"),
+    #                                  ^- available syllabus
+    (1, "Linear Algebra 1", True, 4, None, 3.5, 4, 13, 7),
+    #                                                   ^- complete with all course fields
+    ])
+@pytest.mark.django_db
+def test_course_create_course(valid_courses):
+    flag = True
+
+    # check that valid input doesn't raise exception
+    try:
+        Course(*valid_courses).clean_fields()
+    except ValidationError:
+        flag = False
+
```

Comment:
Kobi suggested in previous reviews that we remove the blank lines before the 'assert', so i think it needs to be removed here as well

Reply:
idk if I need to delete this space line, since I think it adds to readability. Can you show me in which case Kobi said that?

Reply:
Checkout PR #90 and #98 

I don't think its critical - so based on Kobi's comment in these PRs, i think you can implement it in the best way for you.

Reply:
Well, I don't think it's critical either, and I like the separation between the code that gets to the result and the result